### ISSUE TITLE
Loop player acting in an unpredictable fashion

### ISSUE THREAD

**Issue by walt93** [ant-media/Ant-Media-Server#4586]
### Short description
I've created a "livestream" consisting of alternating 12 hour loops - mixed media, alternating with 12 hours of VHS static, emulating a TV station. I'm feeding this livestream to an RTMP via adding an RTMP endpoint.

After leaving this stream running for several days, I observed last night that it was severely out of sync - its mix runs from 12PM->12AM local time - and at 9PM it was playing the VHS static loop - indicating it's at minimum 3 hours out of sync.

I went to restart the stream today on loop 008. I observed it playing the VHS loop.

Somewhat panicked, I stopped the stream, deleted the top two entries, started the stream on loop 009 - and observed it playing the VHS loop again.

I again stopped the stream, I had another copy of the playlist so I deleted the few top entries so we would again start on loop 008, set the RTMP endpoint,  started the stream, and - watched the VHS loop play - again.

After a couple of minutes, it switched over to playing from a video mix. I checked the content to verify where we are in the stream, and it is currently playing loop 012. It should be playing loop 008 - see below.

![Screen Shot 2022-11-17 at 12 23 04 PM](https://user-images.githubusercontent.com/11711050/202551856-297a6f17-3772-40b5-a6d2-783b4ef36b1a.png)

### Environment

* Operating system and version: Ubuntu 20.04.5 LTS
* Java version: openjdk 11.0.17 2022-10-18
* Ant Media Server version: ant-media-server-community-2.5.1.
* Browser name and version: No browser in use in this scenario

### Steps to reproduce
1. Put 25 days of media in the media server, in 12 hour segments
2. Start it playing
3. After several days observe it is out of sync 

## Expected behavior
For it to play the loops consecutively

### Actual behavior
See above

### Logs
__Place logs on [pastebin](http://pastebin.com/) or elsewhere and put links here__

Join [community.antmedia.io](https://community.antmedia.io) and be a part of Ant Media Server Community.

**Comment by yashtandon113** [comment]
Reference Thread in community : https://community.antmedia.io/topic/1478-how-to-controlobserve-where-the-loop-player-is-in-its-playlist/#comment-5332

**Comment by walt93** [comment]
Coming back to this, I've given up tracking down the previous issue, but it hasn't gone away.

I am creating a 24/7/365 podcast, trying to run it in the cloud from Ant Media Player.

Setup is Ant has four identical playlists, all started at the same time. Those playlists are going to one RTMP endpoint each. Those identical playlists are loaded with 12 hour files. I started the four loop players at midnight, so, we are 15 hours into the playlist (e.g. hour 3 of the second file in the playlist).

I have the stream on in the living room, and, it switches to the next loop. I check, and on multiple platforms (e.g. DLive, Rumble, Twitch, Pilled, Conspyre•TV) they have *all* skipped playing file #2 and gone to file #3.

_I quickly stopped/deleted/reloaded/restarted the playlists. Which is a terrible thing to do on a livestream when you have people watching. They tend to not want to watch your stream when you yank their chain like that, but I digress._

I do not believe Ant Media Player is stable enough for my use case. In two weeks of running this stream, Ant has lost the plot multiple times. Good luck fixing your media server. I'm putting a ton of log file in Pastebin. Maybe there is some gold in there for you to mine. https://pastebin.com/KWCEU2ie

**Comment by walt93** [comment]
Update: Reloaded & restarted all four loop players. Today, checking on the streams, they should be (relatively) in sync as they were started within the span of 10-15 seconds.

The streams are anything but in sync. We have 12 hours of media followed by 12 hours of VHS "static" (white noise, indicating we are "off-air"). 

Loop #1 -> OTTfeed - WRONG - is playing "static"
Loop #2 -> restream.io - WRONG - is playing "static"
Loop #3 -> pilled - OK - is playing media
Loop #4 -> rumble - WRONG - is playing "static"

Looking at the playlist URL (I use a CURL command to create these; they are all four identical playlists, differing only in the RTMP endpoint they are streaming to)

https://tv1.s3.us-west-1.wasabisys.com/Conspyre-1.018.mp4
https://s3.us-east-1.wasabisys.com/tv1/12H-VHS-loop.mp4
https://tv1.s3.us-west-1.wasabisys.com/Conspyre-1.019.mp4
https://s3.us-east-1.wasabisys.com/tv1/12H-VHS-loop.mp4

Each of these files are 12 hours long. Maybe something in the referenced log file will tell us why the loop players are skipping out of the file they are playing and advancing to the next prematurely?

https://adserver.conspyre.tv/ant.log.zip            <----- all log files from server, let's see what happened

This may be the last time I can beta test this for you. I am severely screwing up my stream, sending this to an audience. Audiences don't like when you do that to them. I would really like to use Ant for this use case. The alternate is not pretty.

**Comment by walt93** [comment]
That's phenomenal, if a snapshot is no trouble I can install it, let me know what are the logistics.

**Comment by burak-58** [comment]
I sent the Snapshot with a direct message. Did you have chance to test? Could you please write the result here?

**Comment by walt93** [comment]
I had put it in the forums. Copy/pasting here
---
Hi again, I just went through the instructions and it seems my ant media is in a bad state.

Loading from https://kappy.conspyre.tv:5443/ we get the login screen, but the "Let's go" button is dead. JS console has errors:

```
[Debug] AuthService: is authenticated: true local storage: null (main.ca41b26eceb8e3213d33.js, line 1)
[Debug] AuthService navigating login (main.ca41b26eceb8e3213d33.js, line 1)
[Error] Failed to load resource: Could not connect to the server. (first-login-status, line 0)
[Error] Failed to load resource: Could not connect to the server. (z6c3Zzm51I2zB_Gi7146Bg.woff2, line 0)
[Error] Failed to load resource: Could not connect to the server. (bootstrap.min.css.map, line 0)
[Error] Failed to load resource: Could not connect to the server. (authenticate, line 0)
[Error] Failed to load resource: Could not connect to the server. (blocked, line 0)

```

Reloading the page redirects to https://kappy.conspyre.tv:5443/#/pages/login and gets "unable to reach server" from the browser.

I tried installing first with -r true, and again without. No change.

I went through the steps to restore the previous 2.5.1 installation (e.g. stop server, delete antmedia directory, rename previous directory). No change.

Now the above JS errors on login are persistent. Where is additional state stored? I just want to reset this to factory defaults.

**Comment by yashtandon113** [comment]
**Update:** After taking SSH access, the server problem was resolved.

@walt93
Have you had the opportunity to test the Snapshot version? Please let us know if you encounter any problems.

**Comment by asharamseervi** [comment]
Hi @walt93, I can feel your pain! Few takeaways from here:

- Always recommend users to backup before upgrading their version.
- Always brief users for which version/vertical, patch was added while encouraging user for test. Testing instance should be kept separate rather than trying on production, even it's critical fix for end-user.
- Try to adopt Object Storage for content so it didn't vanished away.
- Always git your custom app built from Console, so it just a click away whenever required. _PS: I would say, also git default LiveApp as well as WebRTCAppEE, if you're using these apps._

Thanks.

**Comment by walt93** [comment]
@yashtandon113 yes I'm running it now 24/7 since you helped me with the redeploy - checking it now - we are in time sync. Looks good and thanks!

**Comment by asharamseervi** [comment]
It would be great if you guys share what worked in order to resolve this behavior!

**Comment by yashtandon113** [comment]
@asharamseervi 
Yes, the server update did not go successfully, thus we had to restore the server. The document below explains how to restore a previous version.
https://resources.antmedia.io/docs/upgrade-ams#how-do-i-restore-ant-media-server-if-needed

The playlist problem that @walt93 reported has been fixed in the snapshot version. The Walt's server has been upgraded to the snapshot version after being restored.
